### ISSUE TITLE
UMU support

### ISSUE THREAD

**Issue by foxbat44** [vinegarhq/vinegar#455]
### Before Creating the Enhancement Request

- [x] I have confirmed that this should be classified as an enhancement rather than a bug/feature.

### Is your feature request related to a problem?

Studio on Xwayland currently suffers from an issue where the camera sensitivity is way too high, making it extremely frustrating to use. Neither Gamescope, nor enabling Wine Desktop mitigates this. There have also been reports of white screens and a strange render artifact in the top left corner after opening Studio (see examples below).

![image](https://github.com/user-attachments/assets/12a04dc3-6c45-4ecb-bdfd-1c102389a086)
![image](https://github.com/user-attachments/assets/d39530c9-c7dd-4376-b9e2-6e772646e9b7)

### Describe the Solution you'd Like

Using UMU, which is a launcher for utilizing Proton outside of Steam, seems to fix the aforementioned issues. 

Although initial support was added in version 1.7.0 (old UMU), it seems to have been removed later on as there are only two mentions left of ULWGL inside the master branch, which don't really show any indication for proper support, at least to me. 

For re-adding proper support, I think adding a value for toggling this feature inside the configuration would be a good idea, similar to DXVK. Enabling this feature would also require setting the wineroot path to point towards a Proton build, which the user has to download manually. There **could** also be an option to download this automatically, but maybe that's a bit of a stretch.

### Describe Alternatives you've considered

Right now, using UMU with Studio requires you to download umu-launcher as a system package and set the wineprefix to the one Vinegar uses:

`WINEPREFIX="/home/fxbt/.var/app/org.vinegarhq.Vinegar/data/vinegar/prefixes/studio" GAMEID=0 PROTONPATH="/home/fxbt/GE-Proton9-16" umu-run "/home/fxbt/.var/app/org.vinegarhq.Vinegar/data/vinegar/versions/version-297eb79ede584243/RobloxStudioBeta.exe"`

**Comment by MonkeyDluffySan** [comment]
So if I use UMU the camera problem will be fixed ? For the past week, I am having a bad time running this studio on Linux

**Comment by foxbat44** [comment]
> So if I use UMU the camera problem will be fixed ? For the past week, I am having a bad time running this studio on Linux 

To put it simply, yes.

**Comment by adverbialsatz** [comment]
Everyone who tried umu on XWayland was able to see some improvement.

**Comment by MonkeyDluffySan** [comment]
I switched to Ubuntu on Xorg and all of my issues are gone bruh

**Comment by itsaerosphere** [comment]
> I switched to Ubuntu on Xorg and all of my issues are gone bruh 
> 

Of course the problem disappears when you switch to Xorg. This solution is for people who would rather stay in Wayland, for one reason or another.

Personally, I don't feel like switching sessions just because Studio is the only program where I have issues with Wayland.

This has been a game-changer for me, the mental barrier to dev on Studio has massively decreased. I don't feel like procrastinating as much as before lol

**Comment by apprehensions** [comment]
Why is UMU support required to run Proton? Does setting the Wineroot path to the Proton installation's `files` folder suffice?

```
$ ls
LICENSE                   filelock.py               protonfixes/
LICENSE.OFL               files/                    toolmanifest.vdf
PATENTS.AV1               proton*                   user_settings.sample.py*
compatibilitytool.vdf     proton_3.7_tracked_files  version
$ ls files
bin/   lib/   lib64/ share/
$
```

**Comment by foxbat44** [comment]
> Why is UMU support required to run Proton? Does setting the Wineroot path to the Proton installation's `files` folder suffice?
> 
> ```
> $ ls
> LICENSE                   filelock.py               protonfixes/
> LICENSE.OFL               files/                    toolmanifest.vdf
> PATENTS.AV1               proton*                   user_settings.sample.py*
> compatibilitytool.vdf     proton_3.7_tracked_files  version
> $ ls files
> bin/   lib/   lib64/ share/
> $
> ```

While you can get Studio to run this way, it is not supported and may not work for certain distros: https://github.com/GloriousEggroll/proton-ge-custom#1-running-non-steam-games-with-proton-outside-of-steam-is-not-supported-do-not-ask-for-help-with-this

**Comment by apprehensions** [comment]
That doesn't answer the question i asked. The "Proton" runner is just going to run Wine. It's no different, is this correct?

**Comment by foxbat44** [comment]
From what I've tested, that seems to be the case, though using Vulkan just spits out an error 53.

**Comment by apprehensions** [comment]
Implemented Proton wineroot support (not UMU) in 4ec99a8f7e30459256bf2f737842145b01f82a77
